packages/common/src/api/tan-query/coins/useRedeemCoin.ts

```diff
@@ -0,0 +1,64 @@
+import { Id } from '@audius/sdk'
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+
+import { Feature } from '~/models'
+import { toast } from '~/store/ui/toast/slice'
+
+import { QUERY_KEYS } from '../queryKeys'
+import { useCurrentUserId } from '../users/account/useCurrentUserId'
+import { useQueryContext } from '../utils/QueryContext'
+
+import { getArtistCoinQueryKey } from './useArtistCoin'
+
+type RedeemCoinParams = {
+  mint: string
+}
+
+export const useRedeemCoin = () => {
+  const { audiusSdk, reportToSentry } = useQueryContext()
+  const queryClient = useQueryClient()
+  const { data: currentUserId } = useCurrentUserId()
+
+  return useMutation({
+    mutationFn: async ({ mint }: RedeemCoinParams) => {
+      if (!currentUserId) {
+        throw new Error('User not authenticated')
+      }
+
+      const sdk = await audiusSdk()
+      const response = await sdk.coins.claimCoinReward({
+        mint,
+        userId: Id.parse(currentUserId)
+      })
+
+      return response
+    },
+    onSettled: (_, __, { mint }) => {
```

Comment:
mighttt need optimistic update, probably doesn't matter super much tho?

Reply:
thought of this, but i dont know what i would update. I guess we could optimistically add the reward to the users coin balance, but seems overkill